lib/config.js

```diff
@@ -435,15 +435,13 @@ function normalizeSanitizeFieldNames (opts) {
   }
 }
 
-function normalizeCloudProvider (opts) {
-  const allowedValues = ['auto', 'gcp', 'azure', 'aws', 'none']
-  if (allowedValues.indexOf(opts.cloudProvider) === -1) {
-    opts.cloudProvider = 'auto'
-    if (opts.logger) {
-      opts.logger.warn(
-        'Unexpected cloudProvider value %s, changing to auto',
-        opts.cloudProvider
-      )
+function normalizeCloudProvider (opts, logger) {
+  if ('cloudProvider' in opts) {
+    const allowedValues = ['auto', 'gcp', 'azure', 'aws', 'none']
```

Comment:
I see here we're only normalizing the value if there's an actual property on the `opts` array. My presumption is that it's this line that prevents the the central configuration update from triggering an update of the `cloudProvider` value.